enclave/crypto/CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+if (UNIX OR USE_CLANGW)
+    set(PLATFORM_SRC
+        ../../common/sgx/rand.S)
+elseif (WIN32)
+    set(PLATFORM_SRC
+        ../../common/sgx/rand.asm)
+endif()
+
+add_library(oecrypto STATIC
+    ../../common/asn1.c
+    ../../common/cert.c
+    ../../common/kdf.c
+    asn1.c
+    cert.c
+    crl.c
+    ec.c
+    entropy.c
```

Comment:
Why pull entropy and the RDRAND functions out of oecore into oecrypto? These should be part of the PAL (which is fulfilled by oecore today, and possibly Mike's propose oeposix lib in the future):

 - This avoids the weird circular dependency introduced between mbed*.lib builds and the oecrypto.lib

 - With the introduction of device abstractions and a more fully featured syscall handling, it's likely we would want `/dev/random` or `/dev/urandom` to point to hardware provided entropy fundamental impls, not upwards to oecrypto. 

- It needs to be part of our interface for supporting other crypto libs (i.e. the Coco scenario where they want to do a full mbedtls replacement, or for COSIINE to produce an enclave compatible OpenSSL impl to be used by the OE runtime)

Reply:
`entropy.c` has the `mebdtls_hardware_poll` function, so it's specific to mbedtls. However, I can take the implementation + rd_rand and move it back to oecore. So, theres a platform entropy function. It won't solve the circular dependency however.

Reply:
Specifically for `mbedtls_hardware_poll`, @mikbras and I had a discussion a year back about making mbedtls a fully swappable library without having to change OE runtime code, and one of the options there was to make `mbedtls_hardware_poll` part of the mbedtls build (like we do for oelibc) instead of OE code. i.e. it's the responsibility of the swappable library to integrate the use of entropy from "internal 3rdparty public interfaces" OE provides from the PAL. 

The primary advantage of having `mbedtls_hardware_poll` already available in one of our libraries is only for the case where someone wants to do specifically a direct mbedtls replacement, and I would prioritize a more general encapsulation without the circular reference over catering to that one case.